### PR TITLE
lsfd: use extra information loaded from /proc/net/{raw,raw6,udp6,udplite,udplite6,icmp,icmp6}

### PR DESCRIPTION
**Pull request by masatake**
Close #1835.

### CODE REVIEW

File: misc-utils/lsfd-sock-xinfo.c [hunk 1]
```diff
@@ -1046,3 +1095,170 @@ static void load_xinfo_from_proc_tcp6(ino_t netns_inode)
...
+		   "%x %*x:%*x %*x:%*x %*x %*u %*d %lu ",
+		   local_addr+0, local_addr+1, local_addr+2, local_addr+3, &protocol,
+		   remote_addr+0, remote_addr+1, remote_addr+2, remote_addr+3,
+		   &st, &inode) != 11)
+		return NULL;
+
+	if (inode == 0)
+		return NULL;
+
+	raw = xmalloc(sizeof(struct raw_xinfo));
```
Comment (t-8ch) [inline misc-utils/lsfd-sock-xinfo.c:1205]: `sizeof(*raw)`

File: misc-utils/lsfd-sock-xinfo.c [hunk 2]
```diff
@@ -1046,3 +1095,170 @@ static void load_xinfo_from_proc_tcp6(ino_t netns_inode)
...
+			     char **str)
+{
+	struct raw_xinfo *raw;
+
+	if (l3_fill_column_handler(INET6, sock_xinfo, column_id, str))
+		return true;
+
+	raw = (struct raw_xinfo *)sock_xinfo;
+	if (column_id == COL_RAW_PROTOCOL) {
+		xasprintf(str, "%u", (unsigned int)raw->protocol);
```
Comment (t-8ch) [inline misc-utils/lsfd-sock-xinfo.c:1236]: Could be done without a cast

**Review by t-8ch — APPROVED** [review]

File: misc-utils/lsfd-sock-xinfo.c (hunk 1 above)
Comment (masatake) [reply misc-utils/lsfd-sock-xinfo.c]: Updated. Thank you.

File: misc-utils/lsfd-sock-xinfo.c (hunk 2 above)
Comment (masatake) [reply misc-utils/lsfd-sock-xinfo.c]: Using PRIu16, I removed the case operation. Thank you.

**Comment by t-8ch** [comment]
raw6 tests fail during CI:
```
2023-02-23T09:23:52.5933555Z          lsfd: RAW6 sockets                   ...
2023-02-23T09:23:52.5934560Z diff-{{{
2023-02-23T09:23:52.5940070Z --- /home/runner/work/util-linux/util-linux/tests/expected/lsfd/mkfds-raw6	2023-02-23 09:17:44.521412966 +0000
2023-02-23T09:23:52.5940969Z +++ /home/runner/work/util-linux/util-linux/tests/output/lsfd/mkfds-raw6	2023-02-23 09:23:52.587345868 +0000
2023-02-23T09:23:52.5942258Z @@ -1,2 +1,2 @@
2023-02-23T09:23:52.5943103Z -    3 RAWv6 state=established protocol=5 laddr=::1 raddr=::127.0.0.1 established       raw         ::1 ::127.0.0.1            5
2023-02-23T09:23:52.5943797Z -ASSOC,TYPE,NAME,SOCK.STATE,SOCK.TYPE,INET6.LADDR,INET6.RADDR,RAW.PROTOCOL: 0
2023-02-23T09:23:52.5944338Z +test_mkfds: failed in connect(2): Network is unreachable
2023-02-23T09:23:52.5944961Z +/home/runner/work/util-linux/util-linux/tests/../tests/ts/lsfd/mkfds-raw6: line 45: kill: `': not a pid or valid job spec
2023-02-23T09:23:52.5948365Z }}}-diff
```

**Comment by masatake** [comment]
It seems that "::127.0.0.1" is not usable nowadays.
"::ffff:127.0.0.1" works fine.

**Comment by masatake** [comment]
I want to graft the revised version of #1845 on this request.

**Comment by masatake** [comment]
I pushed all I would like to have in the planned new version.
From the initial commit upto [tests: (lsfd) add a case for UDPLITEv6 sockets](https://github.com/util-linux/util-linux/pull/2079/commits/c934add4be8ecacf5131494bf2d2ca46eb3b0235) I got comments from @t-8ch. 
(Should I add reviewed-by to the commits?) 

After getting the comments, I added (1) code loading /proc/net/{icmp,icmp6} ([tests: (mkfds) add ping and ping6 factories](https://github.com/util-linux/util-linux/pull/2079/commits/00a976118b63df914d80fe04f959345e4ae03151) ~  [tests: (mkfds) add ping and ping6 factories](https://github.com/util-linux/util-linux/pull/2079/commits/00a976118b63df914d80fe04f959345e4ae03151)), and (2) the revised version of #1845 ([tests: (lsfd) add comments about the reason using ts_skip_qemu_user()](https://github.com/util-linux/util-linux/pull/2079/commits/103e235fef8f1f258dca0e77597bd9ce03f11d43) ~ ).

If asked for it, I will split this pull request into smaller ones.

About AF_INET and AF_INET6, I have done all I can do with /proc/net/*.
It means we can add `-i` option that is popular in lsof users.

About the test case of ping, I found a strange behavior.
`test_mkfds ping 3 id=9999` is prevented by SELinux though I run it in unconfined_t.

```
type=AVC msg=audit(1677350610.439:10476): avc:  denied  { name_bind } for  pid=2196482 comm="test_mkfds" src=9999 scontext=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 tcontext=system_u:object_r:port_t:s0 tclass=icmp_socket permissive=0
```

After I did `setenforce 0`, `test_mkfds ping 3 id=9999` runs well.

**Pull request merged**